cmake/dts.cmake

```diff
@@ -175,6 +198,14 @@ if(SUPPORTS_DTS)
     message(FATAL_ERROR "command failed with return code: ${ret}")
   endif()
 
+  #
+  # Add the legacy identifiers
+  #
+
+  file(APPEND ${GENERATED_DTS_BOARD_UNFIXED_H}
```

Comment:
what is this for?

Reply:
Wrote about it on Slack. Seems the bsim tests (part of CI) do something funky where `generated_dts_board.h.deprecated` never gets generated. You get an #include error for it with the old approach.

Reply:
Need to understand this better, I'm not in favor of this solution, and unclear why this makes 'bsim' happy.

Reply:
@galak 
Are you looking into it?

The header doesn't show up, so it breaks otherwise.